### PR TITLE
Create Neural Search Module

### PR DESCRIPTION
**Pull request by zhimin-z**
Intrigued by EdgeGPT, one of the most promising applications of cross-modal machine learning is neural search. Thus I intend to create a new module named "neural search". The core idea of neural search is to leverage state-of-the-art deep neural networks to build every component of a search system. In short, neural search is deep neural network-powered information retrieval. Notably, neural search is not equal to neural architecture search.

Reference:
https://jina.ai/news/what-is-neural-search-and-learn-to-build-a-neural-search-engine
https://medium.com/jina-ai/what-is-jina-and-neural-search-7a9e166608ab

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,8 +20,8 @@ This repository contains a curated list of awesome open source libraries that wi
 | [🗺️ Computation Distribution](#computation-load-distribution) | [📥 Model Serialisation](#model-serialisation) | [🧮 Optimized Computation](#optimized-computation)|
 | [💸 Data Stream Processing](#data-stream-processing) | [:red_circle: Outlier & Anomaly Detection](#outlier-and-anomaly-detection) | [🎁 Feature Store](#feature-store) |
 | [⚔ Adversarial Robustness](#adversarial-robustness) | [💾 Data Storage Optimization](#data-storage-optimisation) | [📓 Data Science Notebook](#data-science-notebook) |
-| [👁️ Industry-strength Computer Vision](#industry-strength-cv) | [🔠 Industry-strength Natural Language Processing](#industry-strength-nlp) | [🍕 Industry-strength Reinforcement Learning](#industry-strength-rl) |
-| [📊 Industry-strength Visualisation](#industry-strength-visualisation) | [💰 Commercial Platform](#commercial-platform) |
+| [🧐 Neural Search](#neural-search) | [👁️ Industry-strength Computer Vision](#industry-strength-cv) | [🔠 Industry-strength Natural Language Processing](#industry-strength-nlp) |
```
Comment (axsaucedo) [inline README.md:23]: Let's add a different emoji for neural search more relevant to the topic

**Review by axsaucedo — CHANGES_REQUESTED** [review]
This is a good idea @zhimin-z - let's add neural search as a section - added a small change and we can then merge

**Comment by axsaucedo** [comment]
Wondering if we can merge (for now) Neural Search & Generative AI as this way we can start growing a section with these, and can then be split once it reaches certain mass

File: README.md (hunk 1 above)
Comment (zhimin-z) [reply README.md:23]: I thought this emoji🧐 this is for (nerual) search, is that not good enough? @axsaucedo

**Comment by zhimin-z** [comment]
> 

There is a question, generative AI is still too broad as a section since it might cover areas from both NLP and CV, then the two sections might not need to exist to some degree. @axsaucedo If we really want to add generative AI, then all modules that are getting involved in generative AI must be split and regenerated.

**Comment by axsaucedo** [comment]
Yes that is a good point, I think it could be worth considering, if they do make sense in a grouping then it could make sense. For the time being I'll merge this, but I would like to continue exploring in the GenAI issue as could be interesting, but would need to make sense.

File: README.md (hunk 1 above)
Comment (axsaucedo) [reply README.md:23]: Oh for me it looks like a concerned face:
image

Maybe it's how it renders, maybe you can use this one 🤓 (or did you mean to use this one?)

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Added reply on discussion

**Review by axsaucedo — APPROVED** [review]
LGTM - thanks Zhimin

**Pull request merged**